src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -342,9 +342,14 @@ void Pane::_CloseChild(const bool closeFirst)
         // Set up new close handlers on the children
         _SetupChildCloseHandlers();
 
-        // Revoke the old event handlers.
-        _firstChild->Closed(_firstClosedToken);
-        _secondChild->Closed(_secondClosedToken);
+        // Revoke the old event handlers on our new children
+        _firstChild->Closed(remainingChild->_firstClosedToken);
+        _secondChild->Closed(remainingChild->_secondClosedToken);
+
+        // Revoke event handlers on old panes and controls
+        oldFirst->Closed(oldFirstToken);
+        oldSecond->Closed(oldSecondToken);
+        closedChild->_control.ConnectionClosed(closedChild->_connectionClosedToken);

```

Comment:
I'm concerned about the mixing of ownership here. Should the child know well enough what it's doing to make its own revocation?

Reply:
That's a good point - it seems like this is the kinda thing that _should_ be handled by the dtor, or maybe the event should be marked `auto_revoke` or something (though I'm not totally sure how that works).

I'd maybe just add a destructor with just
```c++
_control.ConnectionClosed(_connectionClosedToken);
```
<hr>

Though, now that I'm saying that, there is *previous art* on L296-297 where I'm having tha parent manually revoke its children's event handlers on their behalf, so I think I'm fine with this the way it is.

Reply:
Thanks both @DHowett-MSFT @zadjii-msft. Makes sense that this revocation would ideally be done in a destructor. My first aim was simply to prevent the crashing issue, then I attempted to make both branches of the conditional consistent. Only lines 346+347 are actually required to directly resolve the linked issue.

If you like, I can proceed to either:

1. Remove lines 350-352 to strip the changes back to the minimum necessary to fix the crashing issue, then raise a new issue for the ownership question. Or;
2. Move the revocations to a destructor as part of this PR.

Reply:
I'm alright with this, actually. Thank you!